### DISCUSSION TITLE
User Manual for do_simplex_trim

### DISCUSSION THREAD

**Discussion by LFGsim** [JSBSim-Team/jsbsim#840] [Q&A]
Hello,

Happy to be back to JSBSim after more than a decade!
Where can I find details about the different trim cases that can be done with the "**simulation/do_simplex_trim**" function?
For example, how does a "Custom" (option 4) trim is done?

Thank you.
Fabian

**Comment by seanmcleod** [comment]
Do you specifically mean **do_simplex_trim**, since this is the only reference to it in the repo.

https://github.com/JSBSim-Team/jsbsim/search?q=do_simplex_trim

Or do you mean **do_simple_trim**?

https://github.com/JSBSim-Team/jsbsim/search?q=do_simple_trim

https://github.com/JSBSim-Team/jsbsim/blob/375f5be0fdfeb210f4cf66f938e3e6fd426c2966/scripts/737_test.xml#L41-L50

**Comment by LFGsim** [comment]
In fact, my question is for both, do_simplex_trim and do_simple_trim.

**do_simple_trim**:
- Which option should I use for, say, a trim with OEI for straight and level flight. I guess it should be 1: Full, correct?
- What is the option "4: Custom" for?

**do_simplex_trim**:
- What is the difference with do_simple_trim?
- Are the same six options available?

I apologize for so many questions.
Thank you.
Fabián

**Comment by seanmcleod** [comment]
So JSBSim over the years has accumulated 3 different trim implementations, see - https://github.com/JSBSim-Team/jsbsim/issues/227#issuecomment-628597349 however only 1 of them is currently enabled, i.e. the **do_simple_trim** legacy trim routine by Tony Peden.

The **do_simplex_trim** refers to Goppert's trim routines, but it isn't currently enabled. What was added back fairly recently was the linearization code of Goppert.

In terms of the various trim modes for **do_simple_trim** take a look at:

https://github.com/JSBSim-Team/jsbsim/blob/ebb770ebc0ec4cee6aefef6ad88702ef1164b5d7/src/initialization/FGTrim.cpp#L801-L855

**Reply by LFGsim** [reply]
Excellent!
Thank you Sean!

Fabián
